lib/rules/decorator-position.js

```diff
@@ -1,20 +1,29 @@
 'use strict';
 
-const { CLIEngine } = require('eslint');
+const { join } = require('path');
+const { existsSync: isExists, writeFileSync: write, readFileSync: read } = require('fs');
 
-const cli = new CLIEngine();
+const eslintEntrypointFileName = 'eslint-plugin-decorator-position';
+
+if (!isExists(join(findNearestEslintPackage(), `${eslintEntrypointFileName}.js`))) {
+  setupEslintExports();
+}
+
+// eslint-disable-next-line node/no-missing-require
+const { CLIEngine } = require('eslint/eslint-plugin-decorator-position');
```

Comment:
@sviat9440 here is a comparison between our upgrade approaches.

My main question right now:
 - how is your PR passing for eslint 6+7 (and mine not)
 - why do this CLIEngine thing? in eslint 8, we're ok with not having prettier config reading (in favor of manually specifying printWidth)